src/main/java/com/hazelcast/cluster/impl/TcpIpJoinerOverAWS.java

```diff
@@ -17,13 +17,19 @@
 package com.hazelcast.cluster.impl;
 
 import com.hazelcast.aws.AWSClient;
-import com.hazelcast.config.AwsConfig;
+import com.hazelcast.aws.AwsConfig;
 import com.hazelcast.instance.Node;
 import com.hazelcast.logging.ILogger;
 import com.hazelcast.util.ExceptionUtil;
 
 import java.util.Collection;
 
+/**
+ * Joiner used for the deprecated AwsConfig.
+ *
+ * @deprecated Use {@link AwsDiscoveryStrategy} instead of AwsConfig.
+ */
+@Deprecated
```

Comment:
As long as `<aws>` tag uses TcpIpJoinerOverAWS we can't remove it so I think we need to clean-up hazelcast-core.
https://github.com/hazelcast/hazelcast/blob/fcb2de539ae0a6cbe943daa8322505aa48411ddd/hazelcast/src/main/java/com/hazelcast/instance/Node.java#L758

Reply:
Yes, agree. That was actually my meaning of @Deprecated. With the next major hazelcast-aws release, we should remove it both here and in Hazelcast Core.

@mesutcelik Do you think we need to wait for the Hazelcast 4 to clean it up? :) Or the <aws> part does not need to be backward compatible in Hazelcast Core?

Reply:
We were normally putting `@deprecated` in between minor releases. It does not have to be between minor releases. On the other hand, we can consider this TcpIpJoinerOverAWS an internal API so `@deprecated` might not even be needed. It is just a matter of timing for `<aws>` tag be using Discovery SPI under the hood... I think @mmedenjak would be happy if you do the cleanup :)

Reply:
The discussion is really about if we want to be backward compatible with the "<aws>" feature. I'd vote for "no", but let's discuss it separately. If we decide to remove it, I'd be happy to do it both HZ Root and hazelcast-aws repos.

Reply:
Hi guys. I can't remember the exact details but I guess the problem was mostly with removing the `aws` tag (and you mention `TcpIpJoinerOverAWS`). As with removing anything in core, it's a problem. We usually wait for 4.0 to remove API as some customers might be using some old API and we guarantee to be backwards compatible. I can't remember if we even ever removed something after marking it as `@Deprecated`.
But if you think it's really necessary, we might consider it.